### ISSUE TITLE
ACE Heal module not working in MP

### ISSUE THREAD

**Issue by tbeswick96** [oOKexOo/AresModAchillesExpansion#56]
In multiplayer the ACE heal module does not work on other players.
It's a locality issue:
https://github.com/oOKexOo/AresModAchillesExpansion/blob/master/%40AresModAchillesExpansion/addons/modules_f_achilles/ACE/functions/fn_ACEHeal.sqf#L36

```[_x, _x] call ace_medical_fnc_treatmentAdvanced_fullHealLocal;``` 
This needs to be executed where the unit is local. Currently a full heal only works for the play who is also the zeus, but no other units.

```
[_x, _x] remoteExecCall ["ace_medical_fnc_treatmentAdvanced_fullHealLocal", _x, false];
```
This ought to work fine (I used this in a custom module before this was added to this mod, so I know it works properly in multiplayer)

Also, request that the medical modules are moved to the ACE zeus category: 
```category = "ace_zeus_Medical";```

**Comment by oOKexOo** [comment]
Hi tbeswick96

Thx for your comment.

1) A lot of features were initially only executed locally, since units/objects spawned by Zeus are local to him.
I revised most modules by now such that they work for non-local objects as well. It seems that the healing module escaped my notice. I assigned it in #33.

2) ACE category is already adjusted in 0.0.6 dev branch (see #33).

**Comment by tbeswick96** [comment]
@oOKexOo alright sweet good stuff...looking forward to next release

**Comment by comps** [comment]
Note that you don't need to block the rendering for this, so using just `remoteExec` is fine. Also the JIP argument is `false` by default, so you can probably just
```
[_x, _x] remoteExec ["ace_medical_fnc_treatmentAdvanced_fullHealLocal", _x];
```
I'm not sure if using the Local version broadcasts the new medical state to everybody else, we use just `fullHeal` and it works well in MP,
```
    {
        [{
            [player, _this] call ace_medical_fnc_treatmentAdvanced_fullHeal;
        }, _this] call Ares_Extras_fnc_ForUnitsMP;
    }
```

**Comment by freghar** [comment]
Note that you don't need to block the rendering for this, so using just `remoteExec` is fine. Also the JIP argument is false by default, so you can probably just
```
[_x, _x] remoteExec ["ace_medical_fnc_treatmentAdvanced_fullHealLocal", _x];
```
I'm not sure if using the Local version broadcasts the new medical state to everybody else, we use just `fullHeal` and it works well in MP,
```
    {
        [{
            [player, _this] call ace_medical_fnc_treatmentAdvanced_fullHeal;
        }, _this] call Ares_Extras_fnc_ForUnitsMP;
    }
```

**Comment by tbeswick96** [comment]
```ace_medical_fnc_treatmentAdvanced_fullHealLocal``` is preferred over the non-local one as that just calls the local one on the target unit. 
fullHealLocal sets the vars globally. https://github.com/acemod/ACE3/blob/master/addons/medical/functions/fnc_treatmentAdvanced_fullHealLocal.sqf

**Comment by freghar** [comment]
You're right, using the `Local` variant should be fine then.

Regarding scheduled/unscheduled - that depends. In this particular case, it probably doesn't matter very much (several `setVariable`s, some logging calls which are just setting more variables), but - in general - if it's something self-contained that doesn't need external locks, I'd opt for letting the engine scheduler split the runtime between rendered frames (scheduled/`remoteExec`) rather than freezing the game (unscheduled/`remoteExecCall`) as the former will be less gameplay-disrupting (more stable FPS).

**Comment by tbeswick96** [comment]
I'd recommend reading this if you've not already: https://ace3mod.com/wiki/development/arma-3-scheduler-and-our-practices.html
Any function that is significantly large enough to add a visible delay upon being called is probably a badly written function.
Setting some variables won't do anything to the FPS at all. It all executes (and is guaranteed to) in a single frame. If it were run in a scheduled environment, it could be the case that some vars are set a frame after the call, and if the ACE medical functions are reading and acting upon those vars also in a single frame, having only half set could lead to unexpected behaviour.
Anyway, maybe a bit off topic for this issue, as the function call is so lightweight anyway XD

**Comment by oOKexOo** [comment]
I don't entierly agree on the avoid scheduled environment under any circumstance policy. However, it is true that you have to keep in mind the consequences of the different behaviour.
If you like it or not, Ares module code is executed in the scheduled environment which is why for instance dynamic dialogs can be included so easily in the code with a function like `Ares_fnc_showChooseDialog`.

**Comment by oOKexOo** [comment]
Should be resolved in 0.0.6
